File: 53. Maximum Subarray/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int maxi=nums[0],sum=0;
        for(int it:nums)
        {
            sum+=it;
            if(sum>maxi)
                maxi=sum;
            
            if(sum<0)
                sum=0;
        }
        return maxi;
    }
}

File: 53. Maximum Subarray/53. Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        maxi=nums[0]
        sumi=0
        for i in nums:
            sumi+=i
            if(sumi>maxi):
                maxi=sumi
            if sumi<0:
                sumi=0
        return maxi
        